scripts/create_network.py

```diff
@@ -42,10 +31,84 @@ def create_network():
     return n
 
 
-def add_buses_to_network(n, number_microgrids):
-    # Add buses to the network based on the number of microgrids
-    for i in range(number_microgrids):
-        n.madd("Bus", [f"bus{i+1}"], carrier="AC", v_nom=0.220)
+def create_microgrid_network(n, input_file):
+    # Load the GeoJSON file
+    with open(input_file) as f:
+        data = json.load(f)
+
+    # Keep track of the bus coordinates and microgrid IDs
+    bus_coords = set()
+    microgrid_ids = set()
+
+    # Iterate over each feature in the GeoJSON file
+    for feature in data["features"]:
+        # Get the point geometry
+        point_geom = shape(feature["geometry"])
+
+        # Create a bus at the point location with microgrid ID included in bus name
+        bus_name = f"{feature['properties']['microgrid_id']}_bus_{feature['id']}"
+        x, y = point_geom.x, point_geom.y
+
+        # Check for overlapping microgrids and raise an error if happening
+        if (x, y) in bus_coords:
+            raise ValueError(
+                "Overlapping microgrids detected, adjust the coordinates in the config.yaml file"
+            )
+
+        # Add the bus to the network and update the set of bus coordinates and microgrid IDs
+        n.add("Bus", bus_name, x=x, y=y, v_nom=0.220)
```

Comment:
refuse: 0.220?

Reply:
Why? Shouldn't I set a voltage level for the network?